blueocean-core-js/src/js/UrlBuilder.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Created by cmeyers on 8/25/16.
+ */
+
+const extractRestUrl = (subject) => {
+    let restUrl = null;
+
+    if (typeof subject === 'object') {
+        if (subject && subject._links && subject._links.self) {
+            restUrl = subject._links.self.href;
+        }
+    } else if (typeof subject === 'string') {
+        restUrl = subject;
+    }
+
+    if (!restUrl) {
+        throw new Error('Could not find input URL');
+    }
+
+    return restUrl;
+};
+
+const stripEmptyTokens = (tokens) => {
+    const copy = tokens.slice();
+
+    if (copy[0] === '') {
+        copy.shift();
+    }
+
+    if (copy[copy.length - 1] === '') {
+        copy.pop();
+    }
+
+    return copy;
+};
+
+export const isMultiBranchRun = (run) => {
```

Comment:
this is clever (and awful that it's needed), I needed this function many times ;)